epictrack-api/src/api/services/staff.py

```diff
@@ -59,9 +59,19 @@ def find_all_non_deleted_staff(cls, is_active=False):
     @classmethod
     def create_staff(cls, payload: dict):
         """Create a new staff."""
-        exists = cls.check_existence(payload["email"])
+        # Normalize the email and check for existence in the local database
+        email = payload["email"].lower()
+        exists = cls.check_existence(email)
+
         if exists:
             raise ResourceExistsError("Staff with same email already exists")
+        # Fetch user details from Keycloak
+        users = KeycloakService.get_user_by_email(email)
+        if not users:
+            raise ResourceNotFoundError(f"No user found with email: {email}")
```

Comment:
It would be great to have UnprocessbleEntity error here. We are trying to create a staff here, so it won't be good to have ResourceNotFound.